src/components/PlayerSubmissionForm.js

```diff
@@ -3,26 +3,78 @@ import PropTypes from 'prop-types';
 
 import './PlayerSubmissionForm.css';
 
-const PlayerSubmissionForm = () => {
+const PlayerSubmissionForm = (props) => {
+
+  const emptyValues = {};
+  props.fields.forEach((field) => {
+    if (field.key) {
+      emptyValues[field.key] = '';
+    }
+  });
+    
+  const [poemFields, setPoemFields] = useState(emptyValues)
+
+  const onInputChange = (value, name) => {
+    const newFields = {
+      ...poemFields,
+      [name]: value
+    }
+  
+    setPoemFields(newFields);
+  };
+
+  const isValid = (value) => {
+    return value.length > 0
+  }
+  
+  const sendSubmission = (event) => {
+    event.preventDefault();
+    const submission = props.fields.map((field) => {
+      if (field.key) {
+        return poemFields[field.key];
+      } else {
+        return field;
+      } 
+    });
+    props.sendSubmission(submission.join(' '));
+    setPoemFields(emptyValues)
+  }
+
+  const inputValues = props.fields.map((field) => {
```

Comment:
Each item here needs a `key` attribute.  You can wrap the strings in a span like this:

```jsx
else {
  return <span key={field}> {field } </span>
}
```

Making sure each element in the array has a key will get rid of the warning you're seeing.